src/backend/marsha/core/views.py

```diff
@@ -210,6 +210,13 @@ def _get_app_data(self, cache_key, lti=None, resource_id=None):
 
         permissions = {"can_access_dashboard": False, "can_update": False}
 
+        user_id = None
+        if lti:
+            try:
+                user_id = lti.user_id
+            except AttributeError:
+                pass
```

Comment:
```suggestion
        try:
            user_id = lti.user_id if lti else None
        except AttributeError:
            pass
```
?

Reply:
If there is no `user_id` property available in the `lti` object then `user_id` is not set to None. But you are right I should also test if `lti` is not `None`

Reply:
```
        user_id = None
        try:
            user_id = lti.user_id if lti else None
        except AttributeError:
            pass
```

WDYT about that ?

Reply:
can we put it that way ?
```
        try:
            user_id = lti.user_id if lti else None
        except AttributeError:
            user_id = None
```
But it's really a detail, I'm nit-picking here…

Reply:
No that's nice !

Reply:
If you like it, I've got a last shot:
```
user_id = getattr(lti, "user_id", None) if lti else None
```

Reply:
Nice, I can hold it in a tweet now.